### DISCUSSION TITLE
ROS2 Installation

### DISCUSSION THREAD

**Discussion by dave992** [sam-xl/penelope_aerospace_pl_interfaces#1] [Announcements]
A working ROS 2 installation is required to use the action interfaces described in this package. Installation instructions for ROS 2 can be found here: [ROS2 (Humble) Installation Instructions](https://docs.ros.org/en/humble/Installation.html).

I recommend using the Humble (LTS) version of ROS2. Recent versions are available for: Ubuntu, RHEL, and Windows. 

It is also possible to install ROS2 within WSL on Windows 10/11. If you are familiar with containers, Docker is also an option to run and develop your application.

**Update:**
:warning: Multiple partners had issues with installing ROS2 on Windows. If your system can only run Windows, I recommend using RoboStack (see [comment](https://github.com/sam-xl/penelope_aerospace_pl_interfaces/discussions/1#discussioncomment-7058626) from Gijs) or WSL2 instead of the native installation.

**Comment by gavanderhoorn** [comment]
Another alternative could be using [RoboStack](https://github.com/RoboStack/ros-humble) for conda. If you're already familiar with conda for Python packages, this would be the same infrastructure, but used to distribute ROS packages.

Supports Windows natively (without WSL) as well as Linux and macOS.

See [Getting Started](https://robostack.github.io/GettingStarted.html) for how to get started.
